### PR TITLE
New question types

### LINKED ISSUE #40: New Form Features
**Problem:** We are missing some form features the client is looking for
**Solution:** Add support for these features

**Description:**
*Feature:* We need a new type of question that has no answers but is NOT a result. This will represent the end of a section and should be marked as type CONTINUE (instead of RESULT). The buttons should be the same (next, back), and the next button should redirect to another question. This should be represented by the question having one answer field, but this answer field should NOT be displayed, simply used for redirect purposes. 
*Purpose:* User should feel a sense of progress and clean breaks between sections.

*Feature:* We also need support for a tooltip that appears upon hovering. This represents more information about the question that is not necessary to the question itself, but some users may feel the need to gather more information. This will just be represented as an additional field for questions.
*Purpose:* User will be able to hover over some questions to get a more legalese version of question information. This allows the user to understand exactly what and where rules are codified if they wish to do further independent research

**Specs:**
- [x] New question type CONTINUE
  - [x] No visible answer
  - [x] Routing to next and back questions
  - [x] Proof of concept: Add in a CONTINUE with some filler text that occurs after selecting No to cats, and No to cats, and before the question asking why you don't like cats or dogs. This will require new questions and answers in their respective JSONs
- [x] New question fields - tooltip-text, tooltip-hover-text
  - [x] These field should be optional (we don't want an empty tooltip, that's just annoying)
  - [x] If the fields are filled, include the subtext, styled like [this](https://www.figma.com/file/UnjR8tbWQ49DHxHbtLmP7V/Full-Redesign-(All-Breakpoints)?node-id=0%3A1)
  - [x] Upon hovering over the icon, the tooltip should appear above the cursor
  - [x] After moving the mouse off of the icon, the tooltip should disappear
  - [x] Proof of concept: Add in a tooltip for "Do you have a pet?" that says "Do younger siblings count?" that shows "Younger siblings do not count." on hover

**Expected Completion:** 2 weeks
**Resident Expert:** #engineering-support, @dillydally414

### PR DESCRIPTION
**Pull request by clue4**
closes #40

### CODE REVIEW

**Review by dillydally414 — COMMENTED** [review]
Looks pretty good, just had one minor comment and one more major comment. I'm not going to explicitly approve now, because I think we should discuss the major comment and whether that's something worth working on further.

File: constants/questions.json [hunk 1]
```diff
@@ -67,5 +69,15 @@
...
     "type": "RESULT",
     "answers": []
+  },
+  {
+    "id": 6,
+    "question": "You should love animals.",
+    "type": "CONTINUE",
+    "answers": [
+      {
+        "route": 3
```
Comment (devashish9) [inline constants/questions.json:79]: The question with id: 6, does it route back to the question with id: 3? Or are routes and questions connected in some other way?

Comment (dillydally414) [reply constants/questions.json:79]: Yes, it routes back to the question with id: 3. You can view the flowchart representing this question setup [here](https://lucid.app/lucidchart/6e4eaa1f-8f11-44d7-b3e9-a1a7d8e2e369/edit?). If that link doesn't work, go through the Notion: EdLaw > Spring 2022 > External Links: Flowcharts > Animal Form.

File: components/DynamicForm/MyContinue.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React, { useEffect } from 'react'
+import { QuestionText } from '../FormStyles/QuestionText'
+
+interface ContinueProps {
+  label: string
+  onMount: () => void
```
Comment (Nickdaco) [inline components/DynamicForm/MyContinue.tsx:6]: what does onMount do?

File: components/DynamicForm/Tooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import MuiTooltip from '@mui/material/Tooltip'
+import HelpIcon from '@mui/icons-material/Help'
+
```
Comment (Nickdaco) [inline components/DynamicForm/Tooltip.tsx:4]: For clarification, is this file to create pop up text?

**Review by Nickdaco — COMMENTED** [review]
Generally everything looks good. Just needed clarification on a few things I'm not familiar with. Probably wouldn't hurt to have more comments to make it easier to follow each file is doing.

File: components/DynamicForm/Tooltip.tsx (hunk 3 above)
Comment (imanmoreira) [reply components/DynamicForm/Tooltip.tsx:4]: Yup! This component makes the like line of text with the info question mark that you can hover over for the tooltip

File: models/question.ts [hunk 4]
```diff
@@ -11,6 +11,8 @@ export interface Question {
   description?: string
   type: string
   answers: Answer[]
+  tooltipText?: string
```
[the remark this answers is not in the document]
Comment (imanmoreira) [reply models/question.ts:14]: there actually might be a way to abstract it but I wanna think if it makes sense

[the remark this answers is not in the document]
Comment (imanmoreira) [reply models/question.ts:14]: wait jk nah idt it makes sense

[the remark this answers is not in the document]
Comment (imanmoreira) [reply models/question.ts:14]: But i think your tooltip object idea makes sense, I think I like that

File: components/Critical/SideProgressBar.tsx [hunk 5]
```diff
@@ -5,39 +5,47 @@ import ListItem from '@material-ui/core/ListItem'
...
+const StyledListItem = withStyles(
+  {
+    root: {
+      '&.Mui-selected': {
+        background:
+          'linear-gradient(to right, #5064C7 0%, #5064C7 2%, #FFFFFF 2%, #FFFFFF 100%)',
+        boxShadow: '0px 1px 8px rgba(0, 0, 0, 0.15)',
+      },
+      '&:hover': {
+        // repetitive send help
```
[the remark this answers is not in the document]
Comment (clue4) [reply components/Critical/SideProgressBar.tsx]: done

File: models/question.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (clue4) [reply models/question.ts]: done!

File: components/DynamicForm/MyContinue.tsx (hunk 2 above)
Comment (dillydally414) [reply components/DynamicForm/MyContinue.tsx:6]: onMount runs when the component is mounted AKA rendered, and the function they pass in basically updates the formanswer and autoselects the "continue" answer, which isn't actually displayed to the user but still needs to be selected due to the way we handle answers

File: components/FormStyles/FormPageLayout.tsx [hunk 6]
```diff
@@ -2,6 +2,8 @@ import styled from 'styled-components'
 import NavBar from '../Critical/NavBar'
 import SideProgressBar from '../Critical/SideProgressBar'
 
+// THIS IS NEVER USED ANYWHERE (I THINK)
```
Comment (dillydally414) [inline components/FormStyles/FormPageLayout.tsx:5]: does deleting it cause any issues? if not then...

**Review by dillydally414 — APPROVED** [review]
Love this, it can be merged at your earliest convenience

**Pull request merged**